repo/plugin.video.nzbdav/resources/lib/router_search.py

```
@@ -0,0 +1,340 @@
+    """Run provider jobs (serially for one, threaded for many).
+
+    Returns a list of ``(provider_label, (results, error))`` outcomes; a job
+    that raises is surfaced as an empty-results error outcome.
+    """
+    if len(provider_jobs) == 1:
+        provider_key, provider_label, search_func, args, kwargs = provider_jobs[0]
+        try:
+            outcome = _run_one_provider(
+                provider_key, provider_label, search_func, args, kwargs
+            )
+        except Exception as error:  # pylint: disable=broad-exception-caught
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)